test-js/test-utils.js

```diff
@@ -0,0 +1,247 @@
+      expect(iteratorResult.value[0]).equals(0);
+      expect(iteratorResult.value[1]).equals(1);
+    }
+  });
+});
+
+describe("range", () => {
+  it("should generate a range of numbers", () => {
+    const numbers = range(3);
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(0);
+    }
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(1);
+    }
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(2);
+    }
+  });
+  it("should generate a range of numbers, with start and stop values", () => {
+    const numbers = range(10, 13);
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(10);
+    }
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(11);
+    }
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(12);
+    }
+  });
+  it("should generate a range of numbers, with start, stop and step values", () => {
+    const numbers = range(10, 15, 2);
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(10);
+    }
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(12);
+    }
+    {
+      const iteratorResult = numbers.next();
+      expect(iteratorResult.value).equals(14);
+    }
+  });
+});
+
+describe("chain", () => {
+  it("chain iterators", () => {
+    function* numbers() {
+      yield 1;
+      yield 2;
+    }
+    const chained = chain(numbers(), numbers());
+    expect(chained.next().value).equals(1);
+    expect(chained.next().value).equals(2);
+    expect(chained.next().value).equals(1);
+    expect(chained.next().value).equals(2);
+  });
+});
+
+describe("makeUPlusStringFromCodePoint", () => {
+  it("throw an exception when a not valid parameter given", () => {
+    expect(() => makeUPlusStringFromCodePoint("not-a-number")).to.throw();
+  });
+  it("should not throw an exception for a falsy value", () => {
+    expect(() => makeUPlusStringFromCodePoint("")).to.not.throw();
+  });
+  it("make a number unicode hex", () => {
+    expect(makeUPlusStringFromCodePoint(97)).equals("U+0061"); // a
+    expect(makeUPlusStringFromCodePoint(65)).equals("U+0041"); // A
+  });
+});
+
+describe("getCharFromUnicode", () => {
+  it("should return an empty string if an argument is not passed", () => {
+    expect(getCharFromUnicode()).equals("");
+  });
+  it("should convert a unicode symbol number to a character", () => {
+    expect(getCharFromUnicode(97)).equals("a");
+  });
+});
+
+describe("guessCharFromGlyphName", () => {
+  it("should guess a character from code points in free text", () => {
+    expect(guessCharFromGlyphName("text 0061 text")).equals("a");
+    expect(guessCharFromGlyphName("text ff0061 text")).equals("a");
+    expect(guessCharFromGlyphName("text ff0041 text")).equals("A");
```

Comment:
Please add tests with strings containing
- `100000`
- `10FFFF`
- `110000`

Reply:
110000 throws Invalid code point. I tested whether it throws an exception or not, but it feels like this needs to be handled in guessCharFromGlyphName. It should return an empty string instead.

Reply:
Ah yes, I agree this is a bug in `guessCharFromGlyphName`. So at last a new unit test is uncovering a bug :). Feel free to propose a fix.

Reply:
Btw a test for a valid "high" unicode would be good, for example "👀", `U+1F440`.

Reply:
Yes. Hope there's no UTF-16 bug hidden in there...

Reply:
Ok. I will test with naked emoji.